### DISCUSSION TITLE
Sync or async api, how to choose?

### DISCUSSION THREAD

**Discussion by ActivePeter** [WasmEdge/WasmEdge#2955] [General]
Suppose we have many tasks to be executed, one way is to create one vm runtime to and use async api to run tasks in parallel.
Another way is to create many vms to and use multiple threads to run `vm.run_func()`
Which one is better, or is there any trade-off for different workload?

**Comment by hydai** [comment]
It depends on your use cases.

If you would like to isolate every execution environment, using a standalone VM instance is better. If not, you can choose the async one.
